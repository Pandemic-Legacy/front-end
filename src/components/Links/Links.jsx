import React, { useState } from 'react';

import { Grid, Link } from '@material-ui/core';

import { useStyles } from './Links.styles';
import { fetchCountryNameCovidData } from '../../services/covid';
import { setCovidChartData, setSelectedCountryCode } from '../../actions/actions';
import { useDispatch } from 'react-redux';

export const Links = () => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const [searchedCountry, setSearchedCountry] = useState('');

  const handleChange = ({ target }) => {
    setSearchedCountry(target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(setSelectedCountryCode(searchedCountry));
    setCovidChartData(fetchCountryNameCovidData(searchedCountry));
  };

  return (
    <Grid container className={classes.root}>
      <Grid item xs={12} className={classes.links}>
        <form name="country" onSubmit={handleSubmit}>
          <input type="text" value={searchedCountry} onChange={handleChange} className={classes.search} placeholder="Search for country"/>
        </form>      
        <Link className={classes.link} href="/highscore">High Score</Link>
        <Link className={classes.link} href="/">Home</Link>
        <Link className={classes.link} href="/country">Individual Country</Link>
      </Grid>
    </Grid>
  );
};
