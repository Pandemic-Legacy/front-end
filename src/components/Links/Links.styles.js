import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles((theme) => ({
//   root: {
//   },
//   links: {
//     display: 'flex',
//     flexDirection: 'row',
//     justifyContent: 'flex-end',
//     color: theme.palette.primary.main,
//     marginTop: '-4rem',
//     marginRight: '4.5rem',        
//   },
  link: {
    padding: '0 .4rem 0 .4rem',
    // fontWeight: '550',
    fontSize: '1.2rem',
    // color: theme.palette.common.green.main, 
    // '&:hover': {
    //   color: theme.palette.common.blue.main,
    },
    // '&$focusVisible': {
    //   color: theme.palette.common.blue.main,
    // }
  // },
// }
}));
