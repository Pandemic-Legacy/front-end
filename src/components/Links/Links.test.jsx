import React from 'react';
import { shallow } from 'enzyme';
import Links from './Links';
import { ThemeProvider } from '@material-ui/core';

describe('Links component', () => {
  it('renders Links', () => {
    const wrapper = shallow(
      <ThemeProvider>
        <Links />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
