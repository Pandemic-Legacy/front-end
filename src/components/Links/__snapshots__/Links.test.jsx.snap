// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links component renders Links 1`] = `
<ContextProvider>
  <Component />
</ContextProvider>
`;
